app/pulp/app/models/task.py

```diff
@@ -35,6 +35,30 @@ class ReservedResource(Model):
     worker = models.ForeignKey("Worker", on_delete=models.CASCADE, related_name="reservations")
 
 
+class WorkerManager(models.Manager):
+    def get_unreserved_worker(self):
+        """
+        Find an unreserved :class:`pulp.app.models.Worker`
+
+        Return the Worker instance that has no :class:`pulp.app.models.ReservedResource`
```

Comment:
Can Worker receive the linking also.

Note that you can use a ~ to write the full class path, but have Sphinx only show the classname itself.